### PR TITLE
Refac/cas 210 message entity

### PR DESCRIPTION
**Pull request by ogkuzmin**
[CAS-210](https://stream-io.atlassian.net/secure/RapidBoard.jspa?rapidView=49&projectKey=CAS&modal=detail&selectedIssue=CAS-210)

### Description

Refactor repo layer for Messages. Making MessageEntity inner library model, hide from public API.

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: livedata/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt [hunk 1]
```diff
@@ -499,9 +497,7 @@ class ChannelControllerImpl(
                 val processedMessage: Message = result.data()
                 processedMessage.apply {
                     syncStatus = SyncStatus.COMPLETED
-                    val entity = MessageEntity(this)
-                    entity.sendMessageCompletedAt = Date()
```
Comment (ogkuzmin) [inline livedata/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:503]: Check it

File: livedata/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt [hunk 2]
```diff
@@ -608,73 +635,71 @@ internal class ChatDomainImpl private constructor(
...
         pagination: AnyChannelPaginationRequest
-    ): List<ChannelEntityPair> {
+    ): List<Channel> {
         // fetch the channel entities from room
         val channelEntities = repos.channels.select(channelIds)
+        val userIdsFromMessages: Set<String> = if (pagination.memberLimit > 0) {
+            repos.messages.selectUserIdsFromMessagesByChannelsIds(channelIds, pagination)
+        } else {
+            emptySet()
+        }
```
Comment (tschellenbach) [inline livedata/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt:677]: aren't you hitting the database twice with this setup?

Comment (tschellenbach) [reply livedata/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt:677]: once here and then a few lines later with selectMessagesForChannel

**Review by tschellenbach — CHANGES_REQUESTED** [review]
There is a performance regression on the flow for querying channels.

**Review by ogkuzmin — COMMENTED** [review]
Moved quarrying and entity selection to helper. It required create RepositoryFactory (for test purpose). Added tests for that method.

**Review by JcMinarro — APPROVED** [review]
LGTM

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**